src/response.rs

```diff
@@ -0,0 +1,20 @@
+use http::{Response, Uri};
+
+use crate::body::Body;
+use crate::http;
+
+/// Extension trait for http::response::Builder objects
+///
+/// Allows the user to access the `Uri` in http::Response
+pub trait ResponseExt {
+    /// The Uri we ended up at. This can differ from the request uri when we have followed redirects.
+    fn get_uri(&self) -> &Uri;
+}
+
+impl ResponseExt for Response<Body> {
+    fn get_uri(&self) -> &Uri {
+        self.extensions()
+            .get::<Uri>()
```

Comment:
The final question is whether to use a non-public newtype to avoid this becoming a secondary API. You can see another case for that in the same trait, where we pass `Config` in the extensions, but we don't want that object to be a public API via the extensions. So we make this `RequestLevelConfig` wrapper which is `pub(crate)` to lock it down.

https://github.com/algesten/ureq/blob/main/src/config.rs#L563